dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/ingest/tabulardata/impl/plugins/csv/CSVFileReaderTest.java

```diff
@@ -397,15 +404,15 @@ public void testVariableUNFs() {
      * CSVFileReader with a null CSV.
      */
     @Test
-    public void testBrokenCSV() {
+    public void testBrokenCSV() throws IOException {
         try {
             new CSVFileReader(new CSVFileReaderSpi(), ',').read(null, null);
             fail("IOException not thrown on null csv");
         } catch (NullPointerException ex) {
-            assertEquals(null, ex.getMessage());
-        } catch (IOException ex) {
-            String expMessage = BundleUtil.getStringFromBundle("ingest.csv.nullStream");
-            assertEquals(expMessage, ex.getMessage());
+            assertNull(ex.getMessage());
+        } catch (IngestException ex) {
+            IngestError expError = IngestError.UNKNOWN_ERROR;
+            assertEquals(expError, ex.getErrorKey());
```

Comment:
Bonus points for replacing with `@Rule ExpectedException` ;)

Reply:
Hmm, or maybe expected exception in `@Test`? I can't check the IngestError anyway.

Reply:
Why not? In this case you assert for `IngestError.UNKNOWN_ERROR` and lower you assert for `IngestError.RECORD_MISMATCH`. It's already in the code, so not sure what you mean...

Reply:
I mean if I changed the code from `catch (IngestException ex) to ```expectedException.expect(IngestException.class);``` I won't be able to access exception in order to get the Key. I can always assert the message but it was probably not what you meant.

Reply:
Ugh... right, must have been thinking about how Spock handles it nicely... I guess we have Junit5 at least: https://stackoverflow.com/questions/40268446/junit-5-how-to-assert-an-exception-is-thrown Anyways, not a big deal, just `try/catch` looks odd to me in a test.

Reply:
Yeah when writing new tests we are using Junit5 so there we can use Assert.throws. But afair you also cannot access the internal of the exception there.

Reply:
assertThrows() method returns an exception that was throwed